### DISCUSSION TITLE
Unable to locate ipa when using User Interface

### DISCUSSION THREAD

**Discussion by fabolandry** [codemagic-ci-cd/codemagic-docs#1838] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Documentation unclear. No step provided to generate .ipa files

### Expected results

The artifact generated runner.app.zip doesn't contain ipa file. Documentation unclear

### Actual results

runner.ipa.zip uanable to use that file to test the app

### Build id (optional)

64a0c01053936659dce20cfe

**Comment by fabolandry** [comment]
64a0cf21ea591cb8f3740acf New build included --build-name=1.0.$PROJECT_BUILD_NUMBER --build-number=$PROJECT_BUILD_NUMBER. No success still unable to locate .ipa file. documentation very unclear

**Reply by dtrdic** [reply, marked as answer]
Hello Fritz, in order to create a .ipa file and publish it to App Store, you will need to code sign your app first. 
You can check how to set up iOS code signing in the Flutter workflow editor [here](https://docs.codemagic.io/flutter-code-signing/ios-code-signing/)

**Reply by dtrdic** [reply]
You can also check this [step-by-step](https://labs.codemagic.io/your-first-flutter-app-to-appstore/#0) tutorial for publishing your flutter app to App Sotre

**Comment by omar-chaabouni** [comment]
@fabolandry you only get .ipa file when you sign your app
